dapps/W3MWagmi/android/app/build.gradle

```diff
@@ -89,25 +90,56 @@ android {
         versionCode 25
         versionName "1.1"
     }
+
+    def secretsProperties = new Properties()
+    secretsProperties.load(new FileInputStream(rootProject.file("secrets.properties")))
+
     signingConfigs {
         debug {
-            storeFile file('debug.keystore')
-            storePassword 'android'
-            keyAlias 'androiddebugkey'
-            keyPassword 'android'
+            storeFile file(secretsProperties['WC_FILENAME_DEBUG'])
+            storePassword secretsProperties['WC_STORE_PASSWORD_DEBUG']
+            keyAlias secretsProperties['WC_KEYSTORE_ALIAS_DEBUG']
+            keyPassword secretsProperties['WC_KEY_PASSWORD_DEBUG']
+        }
+        internal {
+            storeFile file(secretsProperties['WC_FILENAME_INTERNAL'])
+            storePassword secretsProperties['WC_STORE_PASSWORD_INTERNAL']
+            keyAlias secretsProperties['WC_KEYSTORE_ALIAS']
+            keyPassword secretsProperties['WC_KEY_PASSWORD_INTERNAL']
+        }
+        upload {
+            storeFile file(secretsProperties['WC_FILENAME_UPLOAD'])
+            storePassword secretsProperties['WC_STORE_PASSWORD_UPLOAD']
+            keyAlias secretsProperties['WC_KEYSTORE_ALIAS']
+            keyPassword secretsProperties['WC_KEY_PASSWORD_UPLOAD']
         }
     }
-    buildTypes {
-        debug {
-            signingConfig signingConfigs.debug
+    productFlavors {
+        dev {
+            dimension "version"
+            applicationIdSuffix ".dev"
+            versionNameSuffix "-dev"
+            signingConfig signingConfigs.debug // Use the debug signing config for dev builds
+        }
+        internal {
+            dimension "version"
+            applicationIdSuffix ".internal"
+            versionNameSuffix "-internal"
+            signingConfig signingConfigs.internal
         }
+        upload {
+            dimension "version"
+            signingConfig signingConfigs.upload
+        }
```

Comment:
instead of using flavors, I think it would be better to just create different `buildTypes`.
In Android, the `buildTypes` and flavors` are combined to create multiple variants which is comprised of ever possible combination of buildType and flavor (DevDebug, InternalDebug, UploadDebug, DevRelease, InternalRelease, and UploadRelease)

Instead of using `productFlavors`, create an `internal` buildType instead. [Here's an example](https://www.digitalocean.com/community/tutorials/android-build-types-product-flavors#android-build-types) of how to do it

Reply:
with the current approach i can combine DevDebug, InternalDebug, UploadDebug, DevRelease, InternalRelease, and UploadRelease.

But adding internal as a build type brings problems with some libraries im using, as they apparently only expect `debug` and `release` as buildTypes

Reply:
fallback - https://developer.android.com/build/build-variants#groovy

Reply:
worked perfect 👏